src/nagini_translation/translators/method.py

```diff
@@ -431,6 +431,47 @@ def bind_type_vars(self, method: PythonMethod, ctx: Context) -> None:
                                                                self.no_position(ctx), ctx)
             ctx.bound_type_vars[('V',)] = literal
 
+    def _check_self_type(self, method: PythonMethod, ctx: Context) -> Expr:
+        """
+        Return an expression checking the type of the 'self' variable (only for class methods).
+        """
+        type_check = self.type_factory.type_check(
+                next(iter(method.args.values())).ref(), method.cls, self.no_position(ctx), ctx,
+                concrete=True)
+        return self.viper.Inhale(type_check, self.no_position(ctx),
+                                            self.no_info(ctx))
+
+    def _translate_method_body(self, method: PythonMethod, ctx: Context) -> List[Stmt]:
+        body = []
+        statements = method.node.body
+        body_start, body_end = get_body_indices(statements)
+        # Create local variables for parameters
+        body.extend(self._create_local_vars_for_params(method, ctx))
+        body += flatten(
+            [self.translate_stmt(stmt, ctx) for stmt in
+                method.node.body[body_start:body_end]])    
+        return body
+
+    def _try_block_handlers(self, method: PythonMethod, ctx: Context) -> List[Stmt]:
```

Comment:
For stuff like this, it would be nice to either have a slightly more descriptive name or a docstring (can be super short). Something that tells me that this translates all except and finally blocks in the method.